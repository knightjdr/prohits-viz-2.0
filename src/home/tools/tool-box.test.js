import React from 'react';
import { shallow } from 'enzyme';

import Image from '../../test/test.png';
import scrollTop from '../../helpers/scroll-top';
import ToolBox, { getImageElement } from './tool-box';

// mock ScrollTop
jest.mock('../../helpers/scroll-top');
scrollTop.mockReturnValue();

describe('Toolbox', () => {
  describe('get image element', () => {
    it('should convert image to image element when image is a component', () => {
      const image = <img alt="Toolbox" src="test" />;
      expect(getImageElement(image)).toEqual(image);
    });

    it('should convert image to image element when image is a string', () => {
      const image = <img alt="Toolbox" src="test" />;
      expect(getImageElement('test')).toEqual(image);
    });
  });

  describe('with internal link', () => {
    let wrapper;
    beforeAll(() => {
      scrollTop.mockClear();
      wrapper = shallow(
        <ToolBox
          image={Image}
          route="/test"
          text="Test"
          title="Test"
        />,
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should scroll to top on clicking a nav link', () => {
      wrapper.find('.toolbox__button').simulate('click');
      expect(scrollTop).toHaveBeenCalled();
    });
  });

  describe('with external link', () => {
    let wrapper;
    beforeAll(() => {
      scrollTop.mockClear();
      wrapper = shallow(
        <ToolBox
          external
          image={Image}
          route="/test"
          text="Test"
          title="Test"
        />,
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should not scroll to top on clicking a link', () => {
      wrapper.find('.toolbox__button').simulate('click');
      expect(scrollTop).not.toHaveBeenCalled();
    });
  });
});
