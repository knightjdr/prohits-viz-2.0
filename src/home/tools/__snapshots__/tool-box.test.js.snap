// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbox with external link should match snapshot 1`] = `
<button
  className="toolbox__button"
  onClick={null}
>
  <a
    href="/test"
  >
    <div
      className="toolbox"
    >
      <h4>
        Test
      </h4>
      <div
        className="toolbox__details"
      >
        <p>
          Test
        </p>
        <div
          className="toolbox__image"
        >
          <img
            alt="Toolbox"
            src="test.png"
          />
        </div>
      </div>
    </div>
  </a>
</button>
`;

exports[`Toolbox with internal link should match snapshot 1`] = `
<button
  className="toolbox__button"
  onClick={[Function]}
>
  <Component
    to="/test"
  >
    <div
      className="toolbox"
    >
      <h4>
        Test
      </h4>
      <div
        className="toolbox__details"
      >
        <p>
          Test
        </p>
        <div
          className="toolbox__image"
        >
          <img
            alt="Toolbox"
            src="test.png"
          />
        </div>
      </div>
    </div>
  </Component>
</button>
`;
